src/Conversion/KrnlToAffine/KrnlToAffine.cpp

```diff
@@ -335,6 +344,27 @@ struct AffineBuilder : DialectBuilder {
   void yield() { b.create<AffineYieldOp>(loc); }
 
 private:
+  // Support for multiple forIE loops.
+  void recursionForIE(SmallVectorImpl<IndexExpr> &lbs,
+      SmallVectorImpl<IndexExpr> &ubs, SmallVectorImpl<int64_t> &steps,
+      SmallVectorImpl<Value> &loopIndices,
+      function_ref<void(AffineBuilder &, ValueRange)> builderFn) {
+    int d = loopIndices.size();
+    if (d < (int)lbs.size()) {
+      // Issue a loop and recurse again.
+      forIE(
+          lbs[d], ubs[d], steps[d], [&](AffineBuilder &createAffine, Value i) {
```

Comment:
We should make sure the sizes of `lbs`, `usb`, `step and `loopIndices` are all equal in advance. Just in case one of the values is missing.

Reply:
@tungld The check is done above on line 296 & 297, a function that corresponds to the public interface. This one is called once the check passes, and was needed to handle the recursion (arbitrary number of loop depth). So I think it should be all good

Reply:
I see. You checked them outside. Thanks!